src/back/hlsl/storage.rs

```diff
@@ -366,6 +414,17 @@ impl<W: fmt::Write> super::Writer<'_, W> {
         Ok(())
     }
 
+    /// Set [`temp_access_chain`] to compute the byte offset of `cur_expr`.
+    ///
+    /// The `cur_expr` expression must be a reference to a global
+    /// variable in the [`Storage`] address space, or a chain of
+    /// [`Access`] and [`AccessIndex`] expressions referring to some
+    /// component of such a global.
+    ///
+    /// [`temp_access_chain`]: super::Writer::temp_access_chain
```

Comment:
question: Wait, does this even work for fields? 👀 EDIT: YES IT DOES. 😀 Woot! I'd even been using this notation before, but I don't remember trying it with success before. TIL!